src/repositories/index.ts

```diff
@@ -0,0 +1,44 @@
+import {
+    juggler,
+    DataSourceConstructor,
+    DefaultCrudRepository,
+    Entity,
+} from '@loopback/repository';
+
+const countModel = require('./models/count.model.json');
+
+// I'm unable to do the following ... 
+export class CountRepository extends DefaultCrudRepository<Entity, string> {
+  constructor() {
+    const ds: juggler.DataSource = new DataSourceConstructor({
+        name: 'db',
+        connector: 'memory',
+        file: './data.json',
+    });
+
+    const Count = ds.createModel<typeof juggler.PersistedModel>(
```

Comment:
See http://loopback.io/doc/en/lb4/Thinking-in-LoopBack.html#define-product-model-repository-and-data-source.

The model is defined as a regular class extending from `Entity`, no datasource is involved. The `DefaultCrudRepository` will create datasource-specific model under the hood.

```ts
@model()
export class Count extends Entity {
  @property({type: 'string', id: true, required: true}
  id: string;

  // ...
}
```